lib/models/baseVideo.dart

```diff
@@ -22,17 +22,29 @@ abstract class BaseVideo implements ShareLinks {
   BaseVideo(this.title, this.videoId, this.lengthSeconds, this.author, this.authorId, this.authorUrl, this.videoThumbnails);
 
   @override
-  String getInvidiousLink(Server server) {
-    return '${server.url}/watch?v=$videoId';
+  String getInvidiousLink(Server server, bool isAddTimestamp) {
+    String link = '${server.url}/watch?v=$videoId';
+
+    if (isAddTimestamp) link += '&t=$lengthSeconds';
```

Comment:
```suggestion
    if (timestamp != null) link += '&t=$timestamp';
```

Reply:
Added an `int? timestamp` containing progress in seconds.